ncclient/transport/ssh.py

```diff
@@ -552,21 +501,17 @@ def start_delim(data_len): return '\n#%s\n'%(data_len)
             s.register(chan, selectors.EVENT_READ)
             logger.debug('selector type = %s', s.__class__.__name__)
             while True:
-
-                # Log what netconf:base version we are using this time
-                # round the loop; _base is updated when we receive the
-                # server's capabilities.
-                logger.debug('Currently selected netconf:base:%s', self._base)
-                
+                    
                 # Will wakeup evey TICK seconds to check if something
                 # to send, more quickly if something to read (due to
                 # select returning chan in readable list).
                 events = s.select(timeout=TICK)
                 if events:
                     data = chan.recv(BUF_SIZE)
                     if data:
+                        self._buffer.seek(0, os.SEEK_END)
```

Comment:
Because the underlying buffer is added to by new fragments of the message, and the `_parse11(...)` method moves the pointer back to earlier in the stream to process messages.